Source/WebInspectorUI/UserInterface/Views/ConsoleMessageView.js

```diff
@@ -159,6 +164,43 @@ WI.ConsoleMessageView = class ConsoleMessageView extends WI.Object
         this._repeatCountElement.textContent = Number.abbreviate(count);
     }
 
+    set timestamp(timestamp)
+    {
+        this._message._timestamp = timestamp;
+        if (this._element)
+            this._renderTimestamp(true);
+    }
+
+    _renderTimestamp(renderlatestTimestamp = false)
+    {
+        if (!this._message._timestamp) {
+            if (this._timestampElement) {
+                this._timestampElement.remove();
+                this._timestampElement = null;
+            }
+            return;
+        }
+        if (!this._showTimestamps) {
```

Comment:
A common pattern we use for lists of things that have dynamically showable/hideable content is for the content to always be present in the row (in this case the `ConsoleMessageView`, and let the view of all the rows handle observing the setting and applying a class to the list of console messages that hides/shows all of the timestamps via CSS. This way only one object has to have an observer for the setting changing, instead of every single console message having an observers, of which there might be thousands or more.

If you did this, it would also simplify this method as well as make it easier to understand where style comes from. You would no longer need to check the setting at all in `ConsoleMessageView`, instead making that the responsibility of its parent. It also allows us to implement a cleaner true CSS approach to showing/hiding the timestamps since we can just toggle a class like `shows-timestamps` on the parent view and let CSS handle what gets shown/hidden.

(Note there may still be times where we don't have a timestamp and can't create the element since Web Inspector supports connecting to older backends that may not have the information, so checking if a timestamp exists for the message is still necessary on a row-by-row basis like you did above on :176)